rfd/0167-console-service.md

```diff
@@ -0,0 +1,253 @@
+|`-s,--seconds`|For CPU and trace profiles, profile for the given duration. For other profiles, return a delta profile.|None|
+|`PROFILES`|Comma-separated profile names to be exported. Supported profiles: `allocs`, `block`, `cmdline`, `goroutine`, `heap`, `mutex`, `profile`, `threadcreate`, `trace`|`heap,profile,goroutine`|
+
+Usage examples:
+```bash
+# Output to a file.
+$ teleport debug profile > debug.tar.gz
+
+# Directly extract it to a directory.
+$ teleport debug profile | tar xv -C pprof
+$ ls pprof/
+heap.pprof
+profile.pprof
+goroutine.pprof
+
+# Multiple profiles get merged into a single file.
+$ teleport debug profile heap,goroutine > profile.tar.gz
+```
+
+### Security
+
+Items listed on this section are have their impact limited due to the fact that
+the service will not be exposed outsite the machine/container running the
+Teleport instance.
+
+#### CPU and Memory consumption during profiling
+
+Profiling, especially CPU and memory profiling, can be resource-intensive. While
+capturing these profiles helps diagnose performance bottlenecks, attackers could
+leverage them to launch denial-of-service (DoS) attacks. An attacker could
+consume significant resources by repeatedly triggering profiles, potentially
+slowing down or crashing the Teleport instance.
+
+Given the ease of collecting profiles when using the new service, even regular
+usage can impact the instance. With that being said, we’re going to add rate
+limiting to the profiling endpoints.
+
+In addition, the profiling duraition (sampling) will be limited to the same
+value of the HTTP server write timeout. We'll set this timeout value into fairly
+large value to avoid the ability to diagnose issues.
+
+#### Disk space consumption
+
+Extensive logging (particularly debug-level logs) can consume significant disk
+space. An attacker could fill the disk, impacting not only Teleport but other
+services on the system. To address this, we recommend always turning the log
+level back to what is present on the configuration after the troubleshooting
+session. Adding a predefined timeout to return it automatically could affect the
+debug as there might not be a precise time necessary for the issue to present
+itself.
+
+#### information disclosure
+
+Private keys, tokens, and sensitive ones are unavailable in the result archive.
+However, the information on the profiles might be used to determine the cluster
+state and environment information. It is recommended that users review and
+sanitize the resulting contents.
+
+### Future work
+
+#### Limit command execution to a configured list of users/groups
+
+To limit the access to the `teleport debug` we can introduce a new section
+on the Teleport's configuration that enable users to set a list of permitted
+machine users/groups that can invoke the command.
+
+The service then would need to fetch the PID/UID/GID from the `net.UnixConn`,
+and check if it is on the allowed list.
```

Comment:
As limited as filesystem ACLs are, they would almost surely serve the same purpose here. Besides, aren't the sockets in the datadir, which is already only supposed to be accessible to the user running Teleport?